pkg/controller/cronjob/cronjob_controllerv2.go

```diff
@@ -596,22 +596,32 @@ func (jm *ControllerV2) syncCronJob(
 		klog.ErrorS(err, "Unable to make Job from template", "cronjob", klog.KRef(cronJob.GetNamespace(), cronJob.GetName()))
 		return cronJob, nil, updateStatus, err
 	}
+	jobNotExisted := true
 	jobResp, err := jm.jobControl.CreateJob(cronJob.Namespace, jobReq)
 	switch {
 	case errors.HasStatusCause(err, corev1.NamespaceTerminatingCause):
 	case errors.IsAlreadyExists(err):
 		// If the job is created by other actor, assume  it has updated the cronjob status accordingly
 		klog.InfoS("Job already exists", "cronjob", klog.KRef(cronJob.GetNamespace(), cronJob.GetName()), "job", klog.KRef(jobReq.GetNamespace(), jobReq.GetName()))
-		return cronJob, nil, updateStatus, err
+
+		// When the job created by other actor and the status of cronjob has not been
+		// updated normally, so here need get all jobs again.
+		jobResp, err = jm.jobControl.GetJob(cronJob.GetNamespace(), jobReq.GetName())
```

Comment:
If you want to tackle this case, which looks like a valid one based on your description, I'd suggest the following steps to solve it:
- in this section, when you receive `AlreadyExists` error, you'd need to refresh the object (just like you do here), but additionally you'd need to check owner reference to ensure that the controller is the one who created it. I don't envision us adopting orphaned jobs, that's why the owner ref check is critical for this to work. 

Additionally, I'd like to see a test, which confirms this case is properly covered in both cases:
- pre-existing job with matching owner ref (your fix)
- pre-existing job with non-matching owner ref (current path).

Reply:
Ok, I have added the logic part of check the owner.